Libraries/pbxbuild/Sources/Phase/ProductTypeResolver.cpp

```diff
@@ -335,6 +465,15 @@ resolve(Phase::Environment const &phaseEnvironment, Phase::Context *phaseContext
     }
 
     /*
+     * Create the product's module maps, if required.
+     */
+    if (pbxsetting::Type::ParseBoolean(environment.resolve("DEFINES_MODULE"))) {
```

Comment:
A module map is merely a mapping used to construct the "module".  The module presents a logical "module" which provides a certain set of interfaces (macros, functions, structures, and in the case of C++, classes and templates) which can then be imported as a single unit.  This is also the underlying mechanism behind the interoperability between swift and C.